app/src/main/java/org/sopt/and/presentation/SignUpScreen.kt

```diff
@@ -0,0 +1,184 @@
+    ) {
+        Text(
+            text = "회원가입",
+            color = Color.White,
+            fontSize = 20.sp,
+            textAlign = TextAlign.Center
+        )
+
+        Spacer(modifier = Modifier.height(60.dp))
+
+        Text(
+            text = "이메일과 비밀번호만으로\nWavve를 즐길 수 있어요!",
+            color = Color.White,
+            fontSize = 25.sp,
+            modifier = Modifier.fillMaxWidth()
+        )
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        EmailInputField(value = id, onValueChange = { id = it })
+
+        Text(
+            text = "로그인, 비밀번호 찾기, 알림에 사용되니 정확한 이메일을 입력해주세요.",
+            color = Color.Gray,
+            fontSize = 12.sp,
+            modifier = Modifier.fillMaxWidth()
+        )
+
+        Text(
+            text = "로그인, 비밀번호 찾기, 알림에 사용되니 정확한 이메일을 입력해주세요.",
+            color = Color.Gray,
+            fontSize = 12.sp,
+            modifier = Modifier.fillMaxWidth()
+        )
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        PasswordInputField(
+            value = password,
+            onValueChange = { password = it },
+            passwordVisible = passwordVisible,
+            onPasswordVisibilityChange = { passwordVisible = !passwordVisible }
+        )
+
+        Spacer(modifier = Modifier.height(8.dp))
+
+        Text(
+            text = "비밀번호는 8~20자 이내로 영문 대소문자, 숫자, 특수문자 중 3가지 이상 혼용하여 입력해 주세요.",
+            color = Color.Gray,
+            fontSize = 12.sp,
+            modifier = Modifier.fillMaxWidth()
+        )
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        Text(
+            text = "또는 다른 서비스 계정으로 가입",
+            color = Color.Gray,
+            fontSize = 14.sp,
+            modifier = Modifier.fillMaxWidth(),
+            textAlign = TextAlign.Center
+        )
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        Row(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalArrangement = Arrangement.SpaceEvenly
+        ) {
+            KakaoLoginIcon { /* Kakao 로그인 클릭 시 동작 */ }
+            NaverLoginIcon { /* Naver 로그인 클릭 시 동작 */ }
+            FacebookLoginIcon { /* Facebook 로그인 클릭 시 동작 */ }
+            AppleLoginIcon { /* Apple 로그인 클릭 시 동작 */ }
+        }
+
+        Spacer(modifier = Modifier.weight(1f))
+
+        Column(modifier = Modifier.fillMaxWidth()) {
+            SignUpButton(onClick = {
+                if (validateInput(context, id, password)) {
+                    onSignUpClick()
+                }
+            })
+        }
+    }
+}
+
+private fun validateInput(context: Context, id: String, password: String): Boolean {
+    return when {
+        !Patterns.EMAIL_ADDRESS.matcher(id).matches() -> {
+            showToast(context, "유효한 이메일을 입력하세요.")
+            false
+        }
+
+        !PASSWORD_REGEX.matches(password) -> {
+            showToast(context, "비밀번호는 8~20자 이내로 대소문자, 숫자, 특수문자 조합이어야 합니다.")
+            false
+        }
+
+        else -> true
+    }
+}
```

Comment:
```suggestion
private fun validateInput(context: Context, id: String, password: String): Boolean = when {
    !Patterns.EMAIL_ADDRESS.matcher(id).matches() -> {
        showToast(context, "유효한 이메일을 입력하세요.")
        false
    }
    !PASSWORD_REGEX.matches(password) -> {
        showToast(context, "비밀번호는 8~20자 이내로 대소문자, 숫자, 특수문자 조합이어야 합니다.")
        false
    }
    else -> true
}

```

이렇게 return 없이 사용하는 방법도 있답니다!

Reply:
예시까지,,새로운 정보 감사합니다 :)